### PR TITLE
Minimum node version v18

### PR DESCRIPTION
**Pull request by KristjanESPERANTO**
When the next MagicMirror version is released, node v16 will have reached its end of life.

### CODE REVIEW

File: js/fetch.js [hunk 1]
```diff
@@ -11,15 +11,8 @@
  * @class
  */
 async function fetch(url, options = {}) {
-	// const nodeVersion = process.version.match(/^v(\d+)\.*/)[1];
-	// if (nodeVersion >= 18) {
-	// 	// node version >= 18
-	// 	return global.fetch(url, options);
```
Comment (rejas) [inline js/fetch.js:17]: we cant switch to global.fetch now?

File: .github/workflows/automated-tests.yaml [hunk 2]
```diff
@@ -18,7 +18,7 @@ jobs:
     timeout-minutes: 30
     strategy:
       matrix:
-        node-version: [16.x, 18.x, 20.x]
+        node-version: [18.x, 20.x]
```
Comment (rejas) [inline .github/workflows/automated-tests.yaml:21]: No new version beyond 20 to test against?

File: js/fetch.js (hunk 1 above)
Comment (KristjanESPERANTO) [reply js/fetch.js:17]: I'd love to get rid of the dependency and my calendars also work with global.fetch, but I'm not sure if they all do.

@khassel, you had dealt with it more intensively. What do you think?

File: .github/workflows/automated-tests.yaml (hunk 2 above)
Comment (KristjanESPERANTO) [reply .github/workflows/automated-tests.yaml:21]: No, release date of 21.x is 2023-10-17 :slightly_smiling_face:

File: js/fetch.js (hunk 1 above)
Comment (KristjanESPERANTO) [reply js/fetch.js:17]: I will prepare an extra PR for node-fetch removal. Then we can continue the discussion there. That would have the advantage that we could accept this PR here.

Comment (khassel) [reply js/fetch.js:17]: see https://github.com/MichMich/MagicMirror/pull/2961 and https://github.com/MichMich/MagicMirror/pull/2952

We could try it again and see what happens with the e2e tests, they are meanwhile unstable anyway ...

Comment (KristjanESPERANTO) [reply js/fetch.js:17]: To continue the fetch topic I created PR #3172.

**Review by rejas — APPROVED** [review]

**Pull request merged**